services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/CustomerWithContacts.kt

```diff
@@ -75,15 +81,15 @@ data class Contact(
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonView(ChangeLogView::class)
 data class Customer(
-    val type: CustomerType?, // Mandatory in Allu, but not in drafts.
-    val name: String,
-    val country: String, // ISO 3166-1 alpha-2 country code
-    val email: String?,
-    val phone: String?,
-    val registryKey: String?, // y-tunnus
-    val ovt: String?, // e-invoice identifier (ovt-tunnus)
-    val invoicingOperator: String?, // e-invoicing operator code
-    val sapCustomerNumber: String?, // customer's sap number
+    val type: CustomerType? = null, // Mandatory in Allu, but not in drafts.
+    val name: String? = null,
```

Comment:
Why was name turned into nullable? Why was class changed in the first place?

Reply:
This was done mainly because of new test data simulating the situation when there would not be this data in the json:
```
  "customerWithContacts": {
    "contacts": [],
    "customer": {}
  },
```
If I remove the whole `customerWithContacts` from the json (as it might be done in migration later), then maybe these classes would be okay to remain as they were. But then `ApplicationData.customerWithContacts` needs to be nullable.

Reply:
Or maybe I can use the same test data as there are for the original `ApplicationData`?

Reply:
Why is there test data like that, if it isn't allowed?

Reply:
I got carried away and tried something that is not yet to be implemented, e.g. database migration. That situation in the future when there (maybe) are no customer or contact information in ApplicationData. I will clean up this and use the existing test data and CustomerWithContacts classes.